compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEPadAndBufferize.cpp

```diff
@@ -0,0 +1,260 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-amd-aie/Transforms/PassDetail.h"
+#include "iree-amd-aie/Transforms/Passes.h"
+#include "mlir/Dialect/Bufferization/IR/Bufferization.h"
+#include "mlir/Dialect/Linalg/Transforms/TilingInterfaceImpl.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Dialect/Linalg/Utils/Utils.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Dialect/MemRef/Transforms/Transforms.h"
+#include "mlir/Dialect/SCF/Transforms/TileUsingInterface.h"
+#include "mlir/Dialect/SCF/Transforms/Transforms.h"
+#include "mlir/IR/Iterators.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#define DEBUG_TYPE "iree-amdaie-pad-and-bufferize"
+
+namespace mlir::iree_compiler::AMDAIE {
+
+namespace {
+
+static constexpr StringRef kCopyOpNone = "none";
+
+/// TODO(avarma): For now we are creating a temporary struct options
+/// in order to adapt to BufferizeToAllocation options. Once we address upstream
+/// methods by having APIs, we can better address this instead using the
+/// upstream struct. I could've used the upstream struct here, but I chose
+/// not to in order to first adapt to minimal changes required for the C++
+/// methods to work.
+struct _BufferizeToAllocationOptions {
+  Attribute memorySpace;
+  StringAttr memCpyOp;
```

Comment:
I think apart from memorySpace, all other options can be dropped. We always use the same choice for these.